src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -1311,21 +1303,23 @@ void LedgerConsensusImp::takeInitialPosition (
     }
 
     // Set should be immutable snapshot
-    initialSet = initialSet->snapShot (false);
+    return initialSet->snapShot (false);
+}
 
-    // Tell the ledger master not to acquire the ledger we're probably building
-    ledgerMaster_.setBuildingLedger (previousLedger_->info().seq + 1);
+void LedgerConsensusImp::takeInitialPosition()
+{
+    auto initialSet = makeInitialPosition();
 
-    auto txSet = initialSet->getHash ().as_uint256();
-    JLOG (j_.info()) << "initial position " << txSet;
-    mapCompleteInternal (txSet, initialSet, false);
+    mapCompleteInternal (initialSet, false);
 
     ourPosition_ = std::make_shared<LedgerProposal> (
-        initialLedger->info().parentHash, txSet, closeTime_);
+        previousLedger_->info().hash,
+        initialSet->getHash().as_uint256(),
+        closeTime_);
 
     for (auto& it : disputes_)
     {
-        it.second->setOurVote (initialLedger->txExists (it.first));
+        it.second->setOurVote (initialSet->hasItem (it.first));
```

Comment:
Am I reading this right, this code used to operate on a `Ledger` and now it only operates on the `SHAMap`?